src/qt/walletcontroller.h

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2019 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_QT_WALLETCONTROLLER_H
+#define BITCOIN_QT_WALLETCONTROLLER_H
+
+#include <qt/walletmodel.h>
+#include <sync.h>
+
+#include <list>
+#include <memory>
+#include <vector>
+
+#include <QMutex>
+
+class OptionsModel;
+class PlatformStyle;
+
+namespace interfaces {
+class Handler;
+class Node;
+} // namespace interfaces
+
+/**
+ * WalletController
+ */
+class WalletController : public QObject
```

Comment:
Nit: Wallet**s**Controller?

Reply:
heh 😛avoid plural?

Reply:
Plural makes it clear that this controls multiple wallets, rather than one controller per wallet.

Reply:
I prefer to interpret it as the "gui controller of the wallet related things". Let's see what others say and I'll happily rename.

Reply:
I think either is fine.

Reply:
Tech language prefers a singular form; e.g., "window manager", "file manager" etc.